### PR TITLE
Revisions to ext-mongodb architecture and library tutorial

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPC-2276

### CODE REVIEW

File: reference/mongodb/architecture.xml [hunk 1]
```diff
@@ -10,94 +10,60 @@
...
+    <alt>
+     PHP Driver Architecture Diagram. The lowest level of the driver is our
+     system libraries: libmongoc, libbson, and libmongocrypt. The middle level
+     is the MongoDB PHP C extension. The upper level is PHP userland and
+     includes the MongoDB PHP library and higher-level packages such as
+     framework integrations and applications.
+    </alt>
     <imageobject>
-     <imagedata fileref="en/reference/mongodb/images/driver_arch.png"/>
+     <imagedata fileref="en/reference/mongodb/images/driver_arch.svg" width="625" depth="450" format="SVG" />
```
Comment (jmikola) [inline reference/mongodb/architecture.xml:28]: Replaced the original PNG file with an SVG. There was prior art for this in the PHP manual: https://github.com/php/doc-en/blob/272838c8815f95572a60d30e39dfb1feba3e497c/faq/passwords.xml#L163-L172

Rendered: https://www.php.net/manual/en/faq.passwords.php#faq.password.storing-salts

**Review by jmikola — COMMENTED** [review]
cc: @gromNaN

File: reference/mongodb/architecture.xml [hunk 2]
```diff
@@ -10,94 +10,60 @@
...
-   <link xlink:href="&url.mongodb.libbson;">libbson</link>). This extension
-   will expose an identical public API for the most essential and
-   performance-sensitive functionality:
+   Sitting below that library is a PHP extension, which is distributed through
+   <link xlink:href="&url.pecl.package;mongodb">PECL</link>. The extension forms
+   the glue between PHP and our system libraries
+   (<link xlink:href="&url.mongodb.libmongoc;">libmongoc</link>,
+   <link xlink:href="&url.mongodb.libbson;">libbson</link>, and
+   <link xlink:href="&url.mongodb.libmongocrypt;">libmongocrypt</link>). Its
+   public API provides only the most essential functionality:
```
Comment (Girgias) [inline reference/mongodb/architecture.xml:53]:
```suggestion
   <link xlink:href="&url.pecl.package;mongodb">PECL</link>.
   The extension forms the glue between PHP and our system libraries
   (<link xlink:href="&url.mongodb.libmongoc;">libmongoc</link>,
   <link xlink:href="&url.mongodb.libbson;">libbson</link>, and
   <link xlink:href="&url.mongodb.libmongocrypt;">libmongocrypt</link>).
   Its public API provides only the most essential functionality:
```

File: reference/mongodb/architecture.xml [hunk 3]
```diff
@@ -10,94 +10,60 @@
...
+   At the top of this stack sits a
+   <link xlink:href="&url.mongodb.library;">PHP library</link>, which is
+   distributed as a
+   <link xlink:href="https://packagist.org/packages/mongodb/mongodb">Composer package</link>.
+   This library provides an API consistent with other MongoDB
+   <link xlink:href="&url.mongodb.drivers;">drivers</link> and implements
+   various cross-driver
+   <link xlink:href="&url.mongodb.specs;">specifications</link>. While the
+   extension can be used directly, the library has minimal overhead and should
+   be a common dependency for most applications built with MongoDB.
```
Comment (Girgias) [inline reference/mongodb/architecture.xml:43]:
```suggestion
   At the top of this stack sits a
   <link xlink:href="&url.mongodb.library;">PHP library</link>,
   which is distributed as a
   <link xlink:href="https://packagist.org/packages/mongodb/mongodb">Composer package</link>.
   This library provides an API consistent with other MongoDB
   <link xlink:href="&url.mongodb.drivers;">drivers</link>
   and implements various cross-driver
   <link xlink:href="&url.mongodb.specs;">specifications</link>.
   While the extension can be used directly, the library has minimal overhead and
   should be a common dependency for most applications built with MongoDB.
```

**Review by Girgias — COMMENTED** [review]
Just nits to make the work of translation slightly easier

File: reference/mongodb/architecture.xml [hunk 4]
```diff
@@ -153,7 +106,8 @@
     options). The driver will attempt to find a previously persisted
     <link xlink:href="&url.mongodb.libmongoc;">libmongoc</link> client object for
     that hash. If an existing client cannot be found for the hash, a new client
-    will be created (and persisted for future use).
+    will be created and persisted for future use. This behavior can be disabled
+    via the <literal>"disableClientPersistence"</literal> driver option.
```
[the remark this answers is not in the document]
Comment (alcaeus) [reply reference/mongodb/architecture.xml:116]: Explaining the full behaviour here makes sense 👍

**Review by alcaeus — APPROVED** [review]
Changes LGTM. Thanks @jmikola!

File: reference/mongodb/architecture.xml (hunk 2 above)
Comment (jmikola) [reply reference/mongodb/architecture.xml:53]: @Girgias: Thanks, I'll keep this in mind for future edits.

**Pull request merged**